src/server/server.ts

```diff
@@ -936,8 +912,8 @@ namespace ts.server {
     const options: IoSessionOptions = {
         host: sys,
         cancellationToken,
-        installerEventPort: eventPort,
-        canUseEvents: eventPort === undefined,
+        eventPort,
+        canUseEvents: true,
         useSingleInferredProject,

```

Comment:
Whats the purpose of setting canUseEvents as true all the time?

Reply:
With this change, we can send events regardless of whether eventPort is set or not. If `eventPort === undefined` (ie: sending events through stdin/stdout), the behavior is unchanged. But now tsserver can send events through the event port, so if `eventPort !== undefined`, we can send events as well.

Reply:
I kept the `canUseEvents` option available to avoid breaking the public API. As far as I am aware, @anstarovoyt is the only direct consumer of the IoSession API. Perhaps he can comment on whether `canUseEvents` is a useful option for him. And if not, we can consider deprecating and eventually removing `canUseEvents`.

But we should do this in a separate issue.